lib/migrator.js

```diff
@@ -1,90 +1,88 @@
-  hexo.extend.migrator.register('rss', (args, callback) => {
-    const source = args._.shift();
-    const tomd = new TurndownService();
-
+const request = require('request-promise-native');
+const { parse } = require('url');
+const { readFile } = require('hexo-fs');
+const parseFeed = require('./feed');
+
+module.exports = async function(args) {
+  const source = args._.shift();
+  const tomd = new TurndownService();
+  const { log } = this;
+  const Post = this.post;
+  let untitledPostCounter = 0;
+  let errNum = 0;
+  let input, feed;
+  const posts = [];
+
+  try {
     if (!source) {
       const help = [
         'Usage: hexo migrate rss <source> [--alias]',
         '',
         'For more help, you can check the docs: http://hexo.io/docs/migration.html'
       ];
 
-      console.log(help.join('\n'));
-      return callback();
+      throw help.join('\n');
     }
 
-    const log = hexo.log;
-    const post = hexo.post;
-    let untitledPostCounter = 0;
-    let stream;
-
     if (/^http(s)?:\/\//i.test(source)) {
-      stream = request(source);
+      input = await request(source);
     } else {
-      stream = fs.createReadStream(source);
+      input = await readFile(source);
     }
 
     log.i('Analyzing %s...', source);
 
-    const feedparser = new FeedParser();
-    const posts = [];
-
-    stream.pipe(feedparser)
-      .on('error', callback);
-
-    feedparser.on('error', callback);
-
-    feedparser.on('readable', function() {
-      const stream = this;
-      let item;
-
-      while (item = stream.read()) { // eslint-disable-line
-
-        if (!item.title) {
-          untitledPostCounter += 1;
-          const untitledPostTitle = 'Untitled Post - ' + untitledPostCounter;
-          item.title = untitledPostTitle;
-          log.w('Post found but without any titles. Using %s', untitledPostTitle);
-        } else {
-          log.i('Post found: %s', item.title);
-        }
-
-        const newPost = {
-          title: item.title,
-          date: item.date,
-          tags: item.categories,
-          content: tomd.turndown(item.description || '')
-        };
-
-        if (args.alias) {
-          newPost.alias = url.parse(item.link).pathname;
-        }
+    feed = await parseFeed(input);
+  } catch (err) {
+    throw new Error(err);
+  }
+
+  if (feed) {
+    for (const item of feed.items) {
+      if (!item.title) {
+        untitledPostCounter += 1;
+        const untitledPostTitle = 'Untitled Post - ' + untitledPostCounter;
+        item.title = untitledPostTitle;
+        log.w('Post found but without any titles. Using %s', untitledPostTitle);
+      } else {
+        log.i('Post found: %s', item.title);
+      }
 
-        posts.push(newPost);
+      const newPost = {
+        title: item.title,
+        date: item.date,
+        tags: item.categories,
+        excerpt: tomd.turndown(item.description || ''),
+        content: tomd.turndown(item.content || item.description || '')
```

Comment:
plan to implement a feature to request from the post's url if item.content is absent, in a separate PR.